include/tvm/runtime/device_api.h

```diff
@@ -240,6 +240,10 @@ class TVM_DLL DeviceAPI {
     return device_type != kDLCPU && device_type != kDLMicroDev;
   }
 
+  static bool SupportsPointerArithmetics(int device_type) {
```

Comment:
Since we already have a vtable for `DeviceAPI`, this should be a virtual function instead of a static boolean.  That would allow individual `DeviceAPI` implementations to independently mark that they support the pointer-arithmetic.  (It would also allow checking for driver-dependent support, such as vulkan support for the optional `VK_KHR_buffer_device_address` feature.)

Since host-side pointer arithmetic is not the default behavior for `DLTensor::data`, the default implementation in `DeviceAPI` would return false, and it could be overridden in `CPUDeviceAPI` and `CUDADeviceAPI` to return true.

Reply:
Also, a nitpick: This isn't whether the device supports pointer arithmetic, but whether pointer arithmetic of a device-owned `void* DLTensor::data` may be performed on the host.  The TVM backends for both Vulkan and OpenCL support pointer arithmetic, but only within the generated kernels.  Neither support pointer arithmetic being performed on the host.